pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -281,3 +281,44 @@ func (pa *priorityAssigner) Release(ofPriority uint16) {
 	idxToDel := sort.Search(len(pa.sortedOFPriorities), func(i int) bool { return ofPriority <= pa.sortedOFPriorities[i] })
 	pa.sortedOFPriorities = append(pa.sortedOFPriorities[:idxToDel], pa.sortedOFPriorities[idxToDel+1:]...)
 }
+
+// RevertReassignments reverts priority updates previously computed by the priorityAssigner,
+// if the reassignments failed at the OVS level. Note that it assumes the priortiy updates in
+// the input is not empty and consecutive，e.g. { 100: 99, 99: 98, 98: 97 }
+func (pa *priorityAssigner) RevertReassignments(updates map[uint16]uint16) {
+	siftedUp := false
+	originals := make([]uint16, len(updates))
+	idx := 0
+	for original, updated := range updates {
+		// Use any one of the map entries to determine if the original update was from a siftUp or siftDown operation.
+		if idx == 0 && updated > original {
+			siftedUp = true
+		}
+		originals[idx] = original
+		idx++
+	}
+	// If the original operation was siftUp, the priority reversion should start from the lowest priority
+	// updates and work up to the highest priority updates, in order. This is to ensure that when querying
+	// the original ofPriority, the Priority value was not overridden by other reversion operations.
+	// For reverting siftDown, the rollback will start from highest priority updates instead.
+	if siftedUp {
+		sort.Slice(originals, func(i, j int) bool { return originals[i] <= originals[j] })
+	} else {
+		sort.Slice(originals, func(i, j int) bool { return originals[i] >= originals[j] })
+	}
+	// The Priority corresponding to the first ofPriority in the sorted (or reverse-sorted) original
+	// ofPriorities is always the newly inserted Priority that needs to be released. We do not call
+	// Release here however since the ofPriority will still be occupied after the reversion.
```

Comment:
ditto, I don't get "since the ofPriority will still be occupied after the reversion."
Why would the ofPriority still be occupied? Doesn't L312 release it?

Reply:
L312 deletes the Priority but does not release the ofPriority currently associated with the Priority before reversion. Continuing with the above example, L312 will delete p3, but we know that ofPriority 98 should not be released because there's obviously another Priority that will be taking this ofPriority. In general, the only ofPriority that need to be released is `if _, exists := updates[updated]; !exists`